pycobertura/reporters.py

```diff
@@ -93,152 +67,166 @@ def __init__(self, *args, **kwargs):
-                sources.append((filename, source))
+            sources = [
+                (filename, self.cobertura.file_source(filename))
+                for filename in self.files_info
+            ]
 
         template = env.get_template("html.jinja2")
+        rows = {k: v[:-1] for k, v in lines.items()}
+        footer = {k: v[-1] for k, v in lines.items()}
+
         return template.render(
             title=self.title,
-            lines=formatted_lines[:-1],
-            footer=formatted_lines[-1],
+            lines=rows,
+            footer=footer,
             sources=sources,
             no_file_sources_message=self.no_file_sources_message,
         )
 
 
 class DeltaReporter(object):
-    def __init__(self, cobertura1, cobertura2, show_source=True):
+    def __init__(self, cobertura1, cobertura2, show_source=True, *args, **kwargs):
         self.differ = CoberturaDiff(cobertura1, cobertura2)
+        self.differ_files_info = self.differ.files()
         self.show_source = show_source
+        self.color = kwargs.pop("color", False)
 
-    def get_file_row(self, filename):
-        row_values = [
-            filename,
-            self.differ.diff_total_statements(filename),
-            self.differ.diff_total_misses(filename),
-            self.differ.diff_line_rate(filename),
-        ]
+    @staticmethod
+    def format_line_rate(line_rate):
+        return f"{line_rate:+.2%}" if line_rate else "-"
 
-        if self.show_source is True:
-            missed_lines = self.differ.diff_missed_lines(filename)
-            row_values.append(missed_lines)
-            row = file_row_missed(*row_values)
-        else:
-            row = file_row(*row_values)
+    @staticmethod
+    def format_total_statements(total_statements):
+        return f"{total_statements:+d}" if total_statements else "-"
 
-        return row
+    @staticmethod
+    def format_missed_lines(missed_lines):
+        return [f"+{lno:d}" if is_new else f"-{lno:d}" for lno, is_new in missed_lines]
 
-    def get_footer_row(self):
-        footer_values = [
-            "TOTAL",
-            self.differ.diff_total_statements(),
-            self.differ.diff_total_misses(),
-            self.differ.diff_line_rate(),
-        ]
+    def determine_color_of_number(self, number):
+        return red if number.startswith("+") else green
 
-        if self.show_source:
-            footer_values.append([])  # dummy missed lines
-            footer = file_row_missed(*footer_values)
+    def color_number(self, numbers):
+        if numbers and self.color:
+            if type(numbers) is str:
+                color = self.determine_color_of_number(numbers)
+                result = color(numbers)
+            else:
+                result = ", ".join(
+                    [
+                        self.determine_color_of_number(number)(number)
+                        for number in numbers
+                    ]
+                )
```

Comment:
Is this formatted by Black? Otherwise, I'd suggest having it more compact as such:

```suggestion
                result = ", ".join([
                    self.determine_color_of_number(number)(number)
                    for number in numbers
                ])
```

Reply:
it is indeed formatted by black :)